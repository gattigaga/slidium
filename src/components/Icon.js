import React from "react";
import PropTypes from "prop-types";

const Icon = ({ name, className, onClick }) => (
  <i className={`material-icons ${className || ""}`} onClick={onClick}>
    {name}
  </i>
);

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func
};

export default Icon;
