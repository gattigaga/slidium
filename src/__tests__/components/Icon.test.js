import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Icon from "components/Icon";

let wrapper, onClick;

beforeEach(() => {
  onClick = jest.fn();
  wrapper = shallow(<Icon name="print" onClick={onClick} />);
});

describe("Icon", () => {
  it("renders correctly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders with additional class", () => {
    wrapper.setProps({
      className: "my-class"
    });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("calls 'onClick'", () => {
    expect(onClick.mock.calls.length).toEqual(0);
    wrapper.simulate("click");
    expect(onClick.mock.calls.length).toEqual(1);
  });
});
