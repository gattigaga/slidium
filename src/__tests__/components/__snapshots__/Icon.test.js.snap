// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<i
  className="material-icons "
  onClick={[Function]}
>
  print
</i>
`;

exports[`Icon renders with additional class 1`] = `
<i
  className="material-icons my-class"
  onClick={[Function]}
>
  print
</i>
`;
